### PR TITLE
fix various options definition for ts usage

### PR DESCRIPTION
**Pull request by vincent-tr**
Looking at mqtt-packet, 'will' payload should be a buffer

### CODE REVIEW

File: types/lib/client-options.d.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export interface IClientOptions extends ISecureClientOptions {
     /**
      * the message to publish
      */
-    payload: string
+    payload: Buffer
```
Comment (mcollina) [inline types/lib/client-options.d.ts]: I think it can be both a string or a buffer

**Comment by vincent-tr** [comment]
I've also marked 'qos' as optional on client.publish()

**Comment by vincent-tr** [comment]
Sure, I did not see we have tests on typescript definitions, I will add some.

**Comment by vincent-tr** [comment]
👍 Thanks!

**Comment by vincent-tr** [comment]
Is there any chance to get it released soon

**Pull request merged**